src/wp-includes/block-supports/layout.php

```diff
@@ -883,17 +883,45 @@ function wp_restore_group_inner_container( $block_content, $block ) {
 		return $block_content;
 	}
 
-	$replace_regex   = sprintf(
+	/*
+	 * This filter runs after the layout classnames have been added to the block, so they
+	 * have to be removed from the outer wrapper and then added to the inner.
+	 */
+	$layout_classes = array();
+	$processor      = new WP_HTML_Tag_Processor( $block_content );
+
+	if ( $processor->next_tag( array( 'class_name' => 'wp-block-group' ) ) ) {
+		foreach ( $processor->class_list() as $class_name ) {
+			if ( str_contains( $class_name, 'layout' ) ) {
```

Comment:
This seems a bit error-prone to me. Shouldn't we maybe check for the more specific class names that can be there, or at least for a more specific pattern? What if any custom class name includes the word `layout`? It would be covered here when it probably shouldn't.

Reply:
Good point, the problem is finding one pattern that covers all classnames. We can have:
`is-layout-[type]`
`wp-block-group-is-layout-[type]`
`wp-container-core-group-layout-[random number]`

At most I think we could change the auto-generated class (the one with the random number) to match `is-layout` and check for that. Do you reckon that's worth trying?

Reply:
Yeah, I think that would be a reasonable improvement. Then we could probably look for `is-layout-` (since there's always a suffix if I'm not mistaken)?

Reply:
Done! Yes, there wouldn't ever be a class that ended in `is-layout`.